backend/RoomPlanner/room-planner/src/main/java/edu/roomplanner/repository/ReservationRepository.java

```diff
@@ -0,0 +1,15 @@
+package edu.roomplanner.repository;
+
+import edu.roomplanner.entity.ReservationEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.Calendar;
+import java.util.List;
+
+public interface ReservationRepository extends JpaRepository<ReservationEntity, Long> {
+    @Query(value = "SELECT * FROM reservations p WHERE p.room_id = ?3 and ((p.start_date >= ?1 and p.end_date <= ?2 ) or (p.end_date >= ?1 and p.end_date <= ?2)" +
```

Comment:
When you use 1, 2, 3 for the parameters is hard to read the query, you should use their names.
@Param("startDate") and use the name of the parameter not a number in the query.